conans/test/assets/sources.py

```diff
@@ -120,6 +120,122 @@ def gen_function_cpp(**context):
     return t.render(**context)
 
 
+_function_c = r"""
+
+#define STRINGIFY_(x) #x
+#define STRINGIFY(y) STRINGIFY_(y)
+
+{% if name != "main" %}
+#include "{{name}}.h"
+{% endif %}
+
+#include <stdio.h>
+
+{% for it in includes -%}
+#include "{{it}}.h"
+{% endfor %}
+
+int {{name}}(){
+    #ifdef NDEBUG
+    printf("{{ msg or name }}: Release!\n");
+    #else
+    printf("{{ msg or name }}: Debug!\n");
+    #endif
+
+    // ARCHITECTURES
+    #ifdef _M_X64
+    printf("  {{ msg or name }} _M_X64 defined\n");
+    #endif
+
+    #ifdef _M_IX86
+    printf("  {{ msg or name }} _M_IX86 defined\n");
+    #endif
+
+    #if __i386__
```

Comment:
For completeness, this is missing the cases for `_M_ARM64_` (windows/msvc) and `__aarch64__` (others) that are in the C++ version. In case we ever use the `gen_function_c` on Linux/mac.

Reply:
Done